internal/cli/atlas/teams/rename.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package teams
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	store "github.com/mongodb/mongodb-atlas-cli/internal/store/atlas"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115005/admin"
+)
+
+var renameTemplate = "Team '{{.Name}}' updated.\n"
+
+type renameOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	name   string
+	teamID string
+	store  store.TeamRenamer
+}
+
+func (opts *renameOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *renameOpts) Run() error {
+	r, err := opts.store.RenameTeam(opts.ConfigOrgID(), opts.teamID, opts.newTeam())
+	if err != nil {
+		return err
+	}
+	return opts.Print(r)
+}
+
+func (opts *renameOpts) newTeam() *atlasv2.Team {
+	return &atlasv2.Team{
+		Name: opts.name,
+	}
+}
+
+// RenameBuilder atlas team(s) update <newName> [--orgId orgId].
+func RenameBuilder() *cobra.Command {
+	opts := &renameOpts{}
+	cmd := &cobra.Command{
+		Use:     "rename <newName>",
+		Aliases: []string{"update"},
```

Comment:
just a thought (no change needed), after looking at the `atlas teams create <team> --username=...`, I was thinking initially of not using this alias for rename, as it will block in the future the creation of an `atlas teams update --username=...` command. But saw that for users management we have `atlas teams users` sub-command

Reply:
yes, I did a pass also on the API and this is the update operation but only names are available for update, I'm not sure if this will change in the future and why going with rename as a command but update should also be valid